File: refactor/src/main/java/HelloAction.java
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.CommonDataKeys;
import com.intellij.openapi.editor.Editor;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.Messages;
import org.jetbrains.annotations.NotNull;

public class HelloAction extends AnAction {

    @Override
    public void actionPerformed(AnActionEvent e) {
        // TODO: insert action logic here
        Project project = e.getProject();
        Messages.showMessageDialog(project, "Hello World", "Greeting", Messages.getInformationIcon());
    }

    @Override
    public void update(@NotNull AnActionEvent e) {
        super.update(e);
        final Project project = e.getProject();
        final Editor editor = e.getData(CommonDataKeys.EDITOR);

        boolean visible = false;
        if (project != null && editor != null && editor.getSelectionModel().hasSelection()) {
            visible = true;
        }

        e.getPresentation().setVisible(visible);
    }
}

File: refactor/src/main/kotlin/HelloAction.kt
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.CommonDataKeys
import com.intellij.openapi.ui.Messages

class HelloAction : AnAction() {

    override fun actionPerformed(e: AnActionEvent) {
        // TODO: insert action logic here
        val project = e.project
        Messages.showMessageDialog(project, "Hello World", "Greeting", Messages.getInformationIcon())
    }

    override fun update(e: AnActionEvent) {
        super.update(e)
        val project = e.project
        val editor = e.getData(CommonDataKeys.EDITOR)

        var visible = false
        if (project != null && editor != null && editor.selectionModel.hasSelection()) {
            visible = true
        }

        e.presentation.isVisible = visible
    }
}
